common/src/main/java/com/wynntils/services/mapdata/WaypointsService.java

```diff
@@ -18,18 +18,25 @@
 import com.wynntils.services.mapdata.attributes.type.MapVisibility;
 import com.wynntils.services.mapdata.features.WaypointLocation;
 import com.wynntils.services.mapdata.providers.builtin.MapIconsProvider;
+import com.wynntils.services.mapdata.providers.json.JsonIcon;
 import com.wynntils.services.mapdata.providers.json.JsonMapAttributes;
 import com.wynntils.services.mapdata.providers.json.JsonMapAttributesBuilder;
+import com.wynntils.services.mapdata.providers.json.JsonMapLocation;
 import com.wynntils.services.mapdata.providers.json.JsonMapVisibility;
 import com.wynntils.utils.mc.type.Location;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public class WaypointsService extends Service {
     @Persisted
     private final Storage<List<WaypointLocation>> waypoints = new Storage<>(new ArrayList<>());
 
+    @Persisted
+    private final Storage<List<JsonIcon>> customIcons = new Storage<>(new ArrayList<>());
```

Comment:
Obviously, this is not an ideal location to keep these, but we don't really have working JsonProviders at the moment, and we've agreed with magicus that this is an acceptable porting compromise, that we can fix later. I personally don't want to give up upfixability of these waypoints/icons, which is only possible if we keep them a persisted value (storage/config).